hbase-client/src/main/java/org/apache/hadoop/hbase/client/ZKAsyncRegistry.java

```diff
@@ -229,6 +244,43 @@ private void getMetaRegionLocation(CompletableFuture<RegionLocations> future,
         });
   }
 
+  @Override
+  public CompletableFuture<TableState> getMetaTableState() {
+    return getAndConvert(this.znodeMirroredMetaTableState, ZKAsyncRegistry::getTableState).
+      thenApply(state -> {
+        return state == null || state.equals(ENABLED_META_TABLE_STATE.getState())?
+          ENABLED_META_TABLE_STATE: new TableState(TableName.META_TABLE_NAME, state);
+      }).exceptionally(e -> {
```

Comment:
Currently in HBase, usually we will create a new CompletableFuture and use FutureUtils.addListener to complete it. The code in exceptionally are a bit tricky, where we throw a CompletionException...

Reply:
Yeah, saw that. Here, if no znode -- probably because the mirroring table state manager is turned off -- then I want the return to be ENABLED (Completed CF). Should I change this?

Reply:
Not relevant in new patch